opencga-app/src/main/java/org/opencb/opencga/app/migrations/v3/v3_2_0/TASK_5964/AddNewClinicalStatusValues.java

```diff
@@ -0,0 +1,110 @@
+                        updateDocument.getSet().put("status", newStatus);
+
+                        Document effectiveUpdate = updateDocument.toFinalUpdateDocument();
+
+                        logger.debug("Updating clinical analysis '{}': {}", document.get("id"), effectiveUpdate.toBsonDocument());
+                        bulk.add(new UpdateOneModel<>(Filters.eq("_id", document.get("_id")), effectiveUpdate));
+                    });
+        }
+
+        for (String collection : Arrays.asList(OrganizationMongoDBAdaptorFactory.INTERPRETATION_COLLECTION,
+                OrganizationMongoDBAdaptorFactory.INTERPRETATION_ARCHIVE_COLLECTION,
+                OrganizationMongoDBAdaptorFactory.DELETED_INTERPRETATION_COLLECTION)) {
+            migrateCollection(collection, query, projection,
+                    (document, bulk) -> {
+                        MongoDBAdaptor.UpdateDocument updateDocument = new MongoDBAdaptor.UpdateDocument();
+                        String interpretationId = document.getString("id");
+                        Document status = document.get("status", Document.class);
+
+                        Document newStatus = fillStatusValues("Interpretation", interpretationId, status,
+                                defaultConfiguration.getInterpretation().getStatus());
+                        updateDocument.getSet().put("status", newStatus);
+
+                        Document effectiveUpdate = updateDocument.toFinalUpdateDocument();
+
+                        logger.debug("Updating interpretation '{}': {}", document.get("id"), effectiveUpdate.toBsonDocument());
+                        bulk.add(new UpdateOneModel<>(Filters.eq("_id", document.get("_id")), effectiveUpdate));
+                    });
+        }
+    }
+
+    private Document fillStatusValues(String entity, String id, Document status, List<ClinicalStatusValue> statusValueList) {
+        ClinicalStatus clinicalStatus = new ClinicalStatus();
+        String clinicalId = status != null ? status.getString("id") : null;
+        if (status == null || StringUtils.isEmpty(clinicalId)) {
+            logger.warn("Status is empty or does not contain 'id' field. Setting default status value for {} '{}'", entity, id);
+            for (ClinicalStatusValue clinicalStatusValue : statusValueList) {
+                if (clinicalStatusValue.getType().equals(ClinicalStatusValue.ClinicalStatusType.NOT_STARTED)) {
+                    clinicalStatus.setId(clinicalStatusValue.getId());
+                    clinicalStatus.setDescription(clinicalStatusValue.getDescription());
+                    clinicalStatus.setType(clinicalStatusValue.getType());
+                }
+            }
+        } else {
+            for (ClinicalStatusValue clinicalStatusValue : statusValueList) {
+                if (clinicalStatusValue.getId().equals(clinicalId)) {
+                    clinicalStatus.setId(clinicalStatusValue.getId());
```

Comment:
What if the current study clinical configuration doesn't match the "default" one? 
Is it possible for the clinicalStatus to get a description different from the values on the configuration?
Shouldn't this just transfer the "id" and "description" blindly, and then "translate" the type into the new values?

Reply:
I don't really understand your question. You can only change the status for one that exists in the configuration. OpenCGA will not allow users to change to a non-defined status. Also, users can only set the new status id and everything else (description, type) is filled automatically from the status configuration values.